pkg/kube/watcher.go

```diff
@@ -120,14 +122,21 @@ func WatchForServices(
 
 			eventCh, errorCh, err = watchServices(watchContext, clientset)
 			if err != nil {
-				if isTimeout(err) {
-					// If it's a time out, the server may not be running yet
+				switch {
+				case isTimeout(err):
+					fallthrough
```

Comment:
This might be clearer as:
```golang
switch {
default:
  return err
case isTimeout(err):
case errors.Is(err, unix.ENETUNREACH):
case errors.Is(err, unix.ECONNREFUSED):
case isAPINotReady(err):
}
// Sleep and continue for all the expected errors
time.Sleep(time.Second)
continue
```